### DISCUSSION TITLE
Home Assistant iFrame

### DISCUSSION THREAD

**Discussion by webvision411** [music-assistant/support#1483] [Q&A] [answered]
Hello, I cannot add music assistant to the dashboard in HA since it has to be SSL (https)
/lib/mass doesn't work (404 not found)
The only thing that loads in the iframe is https://hostname/music_assistant_beta/ingress but that has the header and all the extras 
Is there any other way to add it to the dashboard when running home assistant in https?
I'm running version 2.0.0b73

**Comment by OzGav** [comment]
Have you read the FAQs on these two questions? https://github.com/orgs/music-assistant/discussions/710

**Reply by webvision411** [reply]
I did but there's nothing about changing to SSL etc

**Reply by webvision411** [reply]
or why /lib/mass wouldn't be working

**Reply by OzGav** [reply, marked as answer]
From the FAQ 

> Trying to run MA with SSL is not recommended. Having said that, whilst you can not run the stream service behind SSL you can configure the frontend entirely to your requirements. The default is that the frontend is protected by Ingress in HAOS. For those using docker, it is possible to host it on a desired port and then run a (Ingress) reverse proxy. No support will be provided for these setups, we recommend you use HAOS.

/lib/mass was from version 1. You need to enable a port for the front end and accept the security risk that comes from that. By default you will see it is port 8095

**Comment by antialias** [comment]
If Home Aassistant had some URL that would return non-200 when the user is not authenticated it would be very straightforward to use it in a forwardAuth (Traefik) or auth_request (nginx) declaration behind a reverse proxy. I looked pretty hard for such a thing, but the best I could figure out was to put Music Assistant behind a custom service that returned 200 if authenticated, redirected to Home Assistant's oauth login page if not, and implemented a /callback url. Wish I didn't have to do that just to secure a network app on a public port.

**Reply by OzGav** [reply]
This is one of the many reasons we recommend running MA as a HA addon, MA is then secured behind HA authentication system.
